src/Core/Parser/SyntaxTree/AtomNode.php

```diff
@@ -38,7 +38,14 @@ public function evaluate(RenderingContextInterface $renderingContext)
     {
         $atom = clone $renderingContext->getTemplateParser()->parseFile($this->file);
         $arguments = clone $atom->getArguments();
-        $arguments->assignAll((array) $this->getArguments()->getAllRaw() + $renderingContext->getVariableProvider()->getAll())->setRenderingContext($renderingContext);
+        $evaluatedArguments = [];
```

Comment:
I think you can get the same result by:

```php
$arguments->assignAll($this->getArguments()->getArrayCopy() + $renderingContext->getVariableProvider()->getAll())->setRenderingContext($renderingContext);
```

The `getArrayCopy` method will evaluate all arguments against the assigned RenderingContext before returning a plain old array; this is how it differes from `getAllRaw()`.

Reply:
Ok, I'll test that. 
I had also an issue where an argument containing `{foo.{bar}.baz}` did not return the value but an array with parts of that string as values.